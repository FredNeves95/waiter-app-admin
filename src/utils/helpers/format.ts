export const formatToBRCurrency = new Intl.NumberFormat('pt-BR', {
  style: 'currency',
  currency: 'BRL'
});
