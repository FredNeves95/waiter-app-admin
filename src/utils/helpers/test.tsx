import 'jsdom-global/register';
import { mount, shallow,   } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { ReactNode } from 'react';
import { ActivePageProvider } from '../../contexts/ActivePageContext';
import { ModalContextProvider } from '../../contexts/ModalContext';
import { OrderContextProvider } from '../../contexts/OrderContext';

export function mountWithTheme(children: ReactNode) {
  return mount(
    <ThemeProvider theme={theme}>
      <ModalContextProvider>
        <ActivePageProvider>
          <OrderContextProvider>
            {children}
          </OrderContextProvider>
        </ActivePageProvider>
      </ModalContextProvider>
    </ThemeProvider>);
}

export function shallowWithTheme(children: ReactNode) {
  return shallow(
    <ThemeProvider theme={theme}>
      <ModalContextProvider>
        <ActivePageProvider>
          <OrderContextProvider>
            {children}
          </OrderContextProvider>
        </ActivePageProvider>
      </ModalContextProvider>
    </ThemeProvider>);
}
