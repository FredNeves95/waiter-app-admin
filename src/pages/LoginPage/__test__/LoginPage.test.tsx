import {afterEach, beforeEach, describe, expect, it, vi} from 'vitest';
import {LoginPage} from '../index';
import { mountWithTheme } from '../../../utils/helpers/test';
import { BrowserRouter, useNavigate } from 'react-router-dom';


describe('<LoginPage/>', () => {
  beforeEach(() => {
    vi.mock('react-router-dom', async () => {
      const navigate = vi.fn();
      const actual: object = await vi.importActual('react-router-dom');
      return {
        ...actual,
        useNavigate: () => navigate
      };
    });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  const renderComponent = () =>
    mountWithTheme(
      <BrowserRouter>
        <LoginPage/>
      </BrowserRouter>
    );

  it('Should match the snapshot', () => {
    const loginPage = renderComponent();
    expect(loginPage.html()).toMatchSnapshot();
  });

  it('Should render loginPage', () => {
    const loginPage = renderComponent();
    expect(loginPage.isEmptyRender()).toBeFalsy();
  });

  it('Should not execute handleSubmit after clicking the button if there is no email or password', () => {
    const loginPage = renderComponent();

    const handleSubmitSpy = vi.spyOn(console, 'log');
    const button = loginPage.find('button');

    button.simulate('submit');

    expect(handleSubmitSpy).not.toHaveBeenCalled();
  });

  it('Should not execute handleSubmit after clicking the button if there is no valid email', () => {
    const loginPage = renderComponent();

    const navigate = useNavigate();
    const button = loginPage.find('button');

    const emailInput = loginPage.find('input[type="email"]');
    const passwordInput = loginPage.find('input[type="password"]');

    emailInput.simulate('change', {target: {name: 'email', value:'invalid email'}});
    passwordInput.simulate('change', {target: {name: 'password', value:'12345678'}});

    button.simulate('submit');

    expect(navigate).not.toHaveBeenCalled();
  });

  it('Should not execute handleSubmit after clicking the button if there is valid email but password with less than 8 characters', () => {
    const loginPage = renderComponent();

    const navigate = useNavigate();

    const button = loginPage.find('button');

    const emailInput = loginPage.find('input[type="email"]');
    const passwordInput = loginPage.find('input[type="password"]');

    emailInput.simulate('change', {target: {name: 'email', value:'valid@mail.com'}});
    passwordInput.simulate('change', {target: {name: 'password', value:'1234567'}});

    button.simulate('submit');

    expect(navigate).not.toHaveBeenCalled();
  });
  it('Should execute handleSubmit after clicking the button if there is valid email and password', () => {

    const loginPage = renderComponent();
    const button = loginPage.find('button');
    const navigate = useNavigate();
    const emailInput = loginPage.find('input[type="email"]');
    const passwordInput = loginPage.find('input[type="password"]');

    emailInput.simulate('change', {target: {name: 'email', value:'valid@mail.com'}});
    passwordInput.simulate('change', {target: {name: 'password', value:'123456789'}});

    button.simulate('submit');
    expect(navigate).toHaveBeenCalledWith('/home');
  });
});
