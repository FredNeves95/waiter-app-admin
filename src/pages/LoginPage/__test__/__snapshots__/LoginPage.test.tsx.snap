// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LoginPage/> > Should match the snapshot 1`] = `"<div class=\\"sc-jSwlEQ fufEPP\\"><div class=\\"sc-gKHVLF gBDpE\\"><header class=\\"sc-iBAaJG ezEKcw\\"><span class=\\"welcome-text\\">Bem-vindo(a) ao</span><span><strong>WAITER</strong><small>APP</small></span></header><form novalidate=\\"\\" class=\\"sc-ftLKQv bpUXdg\\"><div class=\\"sc-grXZZQ gGSKQt\\"><label class=\\"sc-dkjKgF eTdRzY\\">E-mail<input type=\\"email\\" placeholder=\\"Seu e-mail de acesso\\" name=\\"email\\" error=\\"\\" class=\\"sc-hLclGa bjwZZk\\"></label></div><div class=\\"sc-grXZZQ gGSKQt\\"><label class=\\"sc-dkjKgF eTdRzY\\">Senha<input type=\\"password\\" placeholder=\\"Informe sua senha\\" name=\\"password\\" error=\\"\\" class=\\"sc-hLclGa bjwZZk\\"></label></div><button type=\\"submit\\" disabled=\\"\\" button-type=\\"primary\\" loading=\\"false\\" class=\\"sc-bcPKhP fNBxaM\\">Fazer Login</button></form></div></div>"`;
